### ISSUE TITLE
Add More Colormap Functionality (Set Extrema)

### ISSUE THREAD

**Issue by Hovden** [OpenChemistry/tomviz#306]
There should be a user friendly way in which the user can select the minimum and maximum of the colormap being applied. Right now the user can set these values after clicking an icon and typing values. At our last meeting we discussed a more interactive possibility that is more intuitive to the user.

**Comment by mathturtle** [comment]
This is going to require we pretty much rewrite the color map editor.  I don't think there is a good way to extend the existing one to do this.

**Comment by cryos** [comment]
It will require a new widget, but there is a fair bit we can reuse from the charting that underlies the color map editor. Not a trivial amount of work, but I think it will be worth it.

**Comment by mathturtle** [comment]
If I'm going to be writing a new widget for this, can we get a list of all the things we would like it to do and iterate a bit on the general design before I jump into working on it?

**Comment by cryos** [comment]
Yes, but not today. There are lots of other things to do first anyway.

**Comment by Hovden** [comment]
Is this currently addressed with the "Specify Data Range" button?

**Comment by cryos** [comment]
I would like to add the graphical equivalent to the chart, I really just moved the button closer to the central widget.

**Comment by cquammen** [comment]
I experimented with preliminaries for this, and this is mostly a note to whoever implements this. You may want to override the member functions

`vtkChartHistogram::MouseButtonPressEvent(const vtkContextMouseEvent &m)`
`vtkChartHistogram::MouseButtonReleaseEvent(const vtkContextMouseEvent &m)`

For instance, you may want to set some state that you have selected a data range marker in the button press event, then unset it in the release event. The `MouseButtonPressEvent()` is called okay when you click in the histogram, but `MouseButtonReleaseEvent()` never seems to be called.

I dug a  little to see that `vtkContextScene::ButtonReleaseEvent(const vtkContextMouseEvent &e)` is called, but it never seems to invoke `vtkChartHistogram::MouseButtonReleaseEvent(const vtkContextMouseEvent &m)`. I didn't yet track down why.

**Comment by Hovden** [comment]
If schedule is tight, I'm cool with moving this to next release.
